File: Python/Trainer.py
from FriendlyNPC import FriendlyNPC

class Trainer(FriendlyNPC):
    """NPC pelatih untuk membantu meningkatkan stats player."""
    def __init__(self, name: str, gender: str, race: str, affiliation,
                 is_marryable: bool, birthday: str, haveQuest,
                 fee: int, trainingType: str, **kwargs):
        super().__init__(name, gender, race, affiliation, is_marryable, birthday, haveQuest, **kwargs)
        self.fee = fee
        self.trainingType = trainingType

    def display(self):
        super().display()
        print(f"  Bayaran: {self.fee}")
        print(f"  Tipe Latihan: {self.trainingType}")

File: C++/Trainer.cpp
#pragma once
#include "FriendlyNPC.cpp"

class Trainer : public FriendlyNPC {
private:
    int fee;
    std::string trainingType; // Health / Mana / ActionPoint

public:
    Trainer(std::string name, std::string gender, std::string race, Factions* affiliation,
            bool is_marryable, std::string birthday, Quest* quest,
            int fee, std::string trainingType)
        : Karakter(name, gender, race, affiliation), // Trainer (the most derived) initializes Karakter
          FriendlyNPC(name, gender, race, affiliation, is_marryable, birthday, quest),
          fee(fee), trainingType(trainingType) {}

    // Override Display
    void display() const override {
        FriendlyNPC::display();
        std::cout << "  Bayaran: " << fee << std::endl;
        std::cout << "  Tipe Latihan: " << trainingType << std::endl;
    }
};